Services/Models/SmartDictionary.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
```

Comment:
I don't really  understand why we added this class.  I tlooks like a thin wrapper around ConcurentDictionary with the addition of a Changed prop.  Does it add enough value to keep it as a standalone class?

Reply:
As you noted below, the state and properties are passed by value to the script interpreter. This class, on a get should return a new dictionary. The way the scripts modify the internal state would be through the Set method.

This could be renamed to DeviceProperties and then we just use a ConcurrentDictionary for state, but since they are so similar it made the most sense for the method calls to have a similar pattern.

Reply:
Also important to note that this was reviewed and discussed in https://github.com/Azure/device-simulation-dotnet/pull/173

Due to waiting for the release these look like net new changes